tests/postgres_tests/test_operations.py

```diff
@@ -0,0 +1,143 @@
+
+        project_state = self.set_up_test_model(app_label)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently('Pony', models.Index(fields=['pink'], name='pony_pink_idx'))
+
+        with self.assertRaisesRegex(NotSupportedError, r'cannot be executed inside a transaction'):
+            with connection.schema_editor(atomic=True) as editor:
+                operation.database_forwards(app_label, editor, project_state, new_state)
+
+    def test_add_index_concurrently(self):
+        app_label = 'test_pgcic'
+
+        project_state = self.set_up_test_model(app_label)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently('Pony', models.Index(fields=['pink'], name='pony_pink_idx'))
+        operation.state_forwards(app_label, new_state)
+        self.assertEqual(len(new_state.models[app_label, 'pony'].options['indexes']), 1)
+
+        definition = operation.deconstruct()
+        self.assertEqual(definition[0], 'AddIndexConcurrently')
+        self.assertEqual(definition[1], [])
+        self.assertEqual(definition[2], {
+            'model_name': 'Pony',
+            'index': models.Index(fields=['pink'], name='pony_pink_idx')
+        })
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_forwards(app_label, editor, project_state, new_state)
+
+        self.assertIndexExists('%s_pony' % app_label, ['pink'])
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_backwards(app_label, editor, new_state, project_state)
+
+        self.assertIndexNotExists('%s_pony' % app_label, ['pink'])
+
+    def test_add_brin_index_concurrently(self):
+        """
+        The AddIndexConcurrently operation should support all index types.
+
+        Verify that the operation properly handles the creation of a BRIN index
+        as a spot check, ensuring it does not always use the default (B-tree).
+        """
+        app_label = 'test_pgcbic'
+
+        project_state = self.set_up_test_model(app_label)
+        new_state = project_state.clone()
+
+        operation = AddIndexConcurrently('Pony', BrinIndex(fields=['pink'], name='pony_pink_brin_idx'))
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_forwards(app_label, editor, project_state, new_state)
+
+        self.assertIndexExists('%s_pony' % app_label, ['pink'], index_type='brin')
+
+        with connection.schema_editor(atomic=False) as editor:
+            operation.database_backwards(app_label, editor, new_state, project_state)
+
+        self.assertIndexNotExists('%s_pony' % app_label, ['pink'])
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'RemoveIndexConcurrently is PostgreSQL-specific')
+@modify_settings(INSTALLED_APPS={'append': 'migrations'})
+class RemoveIndexConcurrentlyTests(OperationTestBase):
+    def test_remove_index_concurrently_description(self):
```

Comment:
Remove `_remove_index_concurrently` from test names as it's redundant with the class name.